player/history_heuristic/history_heuristic_node.py

```diff
@@ -0,0 +1,36 @@
+import random
+
+from game_logic.game_status import GameStatus
+from game_logic.igame_state import IGameState
+from player.mcts.imcts_node import IMctsNode
+
+from player.mcts.mcts_node import MctsNode
+
+
+class HistoryHeuristicNode(MctsNode):
+
+    def __init__(self, game_state: IGameState, action: int, parent: IMctsNode, epsilon = 0):
+        super().__init__(game_state, action, parent)
+        self._epsilon = epsilon
+
+    @property
+    def epsilon(self) -> float:
+        return self._epsilon
+
+    def simulate(self, state):
+        actions_history = self.get_actions_history_dictionary()
+        while state.game_status == GameStatus.InProgress:
+            p = random.random()
+            if p < self.epsilon:
+                # Get first available move based on actions history
+                for action in actions_history:
```

Comment:
what's happening if there is no available moves?

Reply:
It means board is fully filled, so game is no longer in progress and iteration of main loop ends.